File: 90-subsets-ii/90-subsets-ii.java
class Solution {
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        Arrays.sort(nums);
        List<List<Integer>>res=new ArrayList<>();
        solve(0,nums,res,new ArrayList<>());
        return res;
    }
    
    public void solve(int ind,int []nums,List<List<Integer>>res,List<Integer>temp){
        res.add(new ArrayList<>(temp));
        for(int i=ind;i<nums.length;i++){
            if(i!=ind && nums[i]==nums[i-1])continue;
            temp.add(nums[i]);
            solve(i+1,nums,res,temp);
            temp.remove(temp.size()-1);
        }
    }
}

File: 90-subsets-ii/90-subsets-ii.cpp
class Solution {
public:
    vector<vector<int>>res;
    
    void solve(int ind,int n,vector<int>&nums, vector<int> &temp){
        res.push_back(temp);
        for(int i=ind;i<n;i++){
            if(i!=ind && nums[i-1]==nums[i])continue;
            temp.push_back(nums[i]);
            solve(i+1,n,nums,temp);
            temp.pop_back();
        }
    }
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        int n=nums.size();
        vector<int>temp;
        sort(nums.begin(),nums.end());
        solve(0,n,nums,temp);
        return res;
    }
};